dashboard/app/views/levels/multi.js

```diff
@@ -0,0 +1,305 @@
+  // If this button is already crossed, do nothing more.
+  if (this.crossedAnswers.indexOf(index) !== -1)
+  {
+    return;
+  }
+
+  // If single answer, and this button is already selected, do nothing more.
+  if (this.numAnswers == 1 && $.inArray(index, this.selectedAnswers) !== -1)
+  {
+    return;
+  }
+
+  // If multiple answer, and this button is already selected, deselect it.
+  if (this.numAnswers > 1 && $.inArray(index, this.selectedAnswers) !== -1)
+  {
+    this.unclickItem(index);
+    return;
+  }
+
+  this.enableButton(true);
+
+  this.lastSelectionIndex = index;
+
+  // Unchecked->checked.
+  $("#" + this.id + " #unchecked_" + index).hide();
+  $("#" + this.id + " #checked_" + index).show();
+
+  // Add this answer to the list of selected answers.
+  this.selectedAnswers.unshift(index);
+
+  // Unselect previously selected answer if there are now too many selected.
+  if (this.selectedAnswers.length > this.numAnswers)
+  {
+    var unselectIndex = this.selectedAnswers.pop();
+
+    // Although don't uncheck it if it's already crossed out.
+    if (this.crossedAnswers.indexOf(unselectIndex) === -1 )
+    {
+      $("#" + this.id + " #unchecked_" + unselectIndex).show();
+      $("#" + this.id + " #checked_" + unselectIndex).hide();
+    }
+  }
+
+  return true;
+};
+
+Multi.prototype.unclickItem = function(index)
+{
+  var selectedItemIndex = this.selectedAnswers.indexOf(index);
+  this.selectedAnswers.splice(selectedItemIndex, 1);
+
+  // Checked->unchecked.
+  $("#" + this.id + " #unchecked_" + index).show();
+  $("#" + this.id + " #checked_" + index).hide();
+};
+
+// called on $.ready
+Multi.prototype.ready = function()
+{
+  // Are we read-only?  This can be because we're a teacher OR because an answer
+  // has been previously submitted.
+  if (window.appOptions.readonlyWorkspace)
+  {
+    // hide the Submit buttons.
+    $("#" + this.id +'.submitButton').hide();
+
+    // grey out the marks
+    $("#" + this.id +'.item-mark').css('opacity', 0.5);
+
+    this.submitAllowed = false;
+
+    // Are we a student viewing their own previously-submitted work?
+    if (window.appOptions.submitted)
+    {
+      // show the Unsubmit button.
+      $("#" + this.id +'.unsubmitButton').show();
+    }
+  }
+
+  $("#" + this.id + " span.answerbutton").click($.proxy(function(event) {
+    //console.log("answerbutton clicked", this.id);
+    this.choiceClicked($(event.currentTarget));
+  }, this));
+
+  $("#" + this.id + ' #voteform img').on('dragstart', $.proxy(function(event) {
+    // Prevent button images from being dragged, click the button instead.
+    var button = $(event.currentTarget).parent().parent().parent();
+    this.choiceClicked(button);
+    event.preventDefault();
+    event.stopPropagation();
+  }, this));
+
+  this.enableButton(false);
+
+  // Pre-select previously submitted response if available.
+  if (this.lastAttemptString)
+  {
+    var previousResult = this.lastAttemptString.replace(/[\[\]]/g, "").split(',');
```

Comment:
This isn't too bad but I I don't like mysterious regexes... would it be possible to format lastAttemptString in JSON for passing between ruby and js?